### ISSUE TITLE
branch aosp7.0: list of bootloop issues

### ISSUE THREAD

**Issue by tank0412** [zf5/android_device_asus_ctp-common#2]
**Issue № 1**: Multidisplay issue
_Descriprion_: Surfaceflinger did not call multidisplay stuff (patch was applyed; You can fing it in the Patches repo. Branch: aosp7.0)
_Logcat:_
```
08-24 09:52:37.677   256   256 W hwcomposer: _drmModeConnector* IntelHWComposerDrm::getConnector(int): fail to get required connector
08-24 09:52:37.677   256   256 W hwcomposer: bool IntelHWComposerDrm::detectDisplayConnection(int): fail to get drm connector
```

**Issue № 2**: (?) issue
_Logcat:_
```
08-24 09:52:38.973   323   323 F libc    : CANNOT LINK EXECUTABLE "/system/bin/mediaserver": cannot locate symbol "_ZN7android8hardware14ICameraService10descriptorE" referenced by "/system/lib/libcameraservice.so"...
F libc    : CANNOT LINK EXECUTABLE "/system/bin/app_process": cannot locate symbol "_ZTVN7android8hardware10CameraInfoE" referenced by "/system/lib/libandroid_runtime.so"...
08-24 09:52:43.693   417   417 F libc    : CANNOT LINK EXECUTABLE "/system/bin/drmserver": cannot locate symbol "_ZN7android8hardware7ICamera11asInterfaceERKNS_2spINS_7IBinderEEE" referenced by "/system/lib/libmedia.so"...
```
Full logcat is there:
[log.txt](https://github.com/zf5/android_device_asus_ctp-common/files/434473/log.txt)

Device tree:
Check repositories in the zf5 organsitaion. Branch: aosp7.0

**Comment by dgadelha** [comment]
**Issues:**
- Multidisplay **(needed to boot)**
- Camera
- libdrm **(needed to boot)**
- mediaserver
- drmserver
- gpsd **(to be removed)**
- syscalls on kernel **(?)**

**Comment by hungximbex** [comment]
AOSP android 7.0 for zenfone 5 bootloop ( 29/9/2016)

**Comment by hungximbex** [comment]
AOSP android 7.0 v2 for zenfone 5 bootloop , I think that Multdisplay.patch not work

**Comment by dgadelha** [comment]
I can confirm that multidisplay.patch works as we had bootanimation.
But our issue was with BatteryService, it was failing to start.

**Comment by tank0412** [comment]
@hungximbex  why you report something WITHOUT **logcat**?

@dgadelha Multidisplay patch does not work.
I had same issue on RR MM builds. You can see a bootanimation without any problems but it will not if logcat has this lines:
```
I ServiceManager: Waiting for service display.intel.mds...
E MultiDisplay: drmModeConnector *getConnector(int, uint32_t): Failed to get conector
```

**Comment by dgadelha** [comment]
DP2 is a Marshmallow system.
The only nougat manifest is 7.0_rX
Trust me, I've already built and tested the dev previews.

**Comment by tank0412** [comment]
@dgadelha  Yeah N DP2 which i boot was very similar to Marshmallow but it used Noughat compilator and etc.

**Comment by dgadelha** [comment]
It is a Marshmallow with Ninja.
Only that.

**Comment by hungximbex** [comment]
AOSP android 7.0 for zenfone 5 ( 30/9) Bootloop

**Comment by hungximbex** [comment]
@tank0412 , AOSP ( 30/9) for zenfone 5 bootloop

**Comment by tank0412** [comment]
Well, i updated selinux in the kernel.
https://github.com/zf5/android_kernel_asus_T00F/commits/cm-14.0
We became a bit nearer for boot! xD
But still is not near enough.
```
W PackageManager: Destroying /data/user/0/emmc_total_size due to: com.android.server.pm.PackageManagerException: Package emmc_total_size is unknown
E : Couldn't opendir /data/data/emmc_total_size: Not a directory
E SELinuxMMAC: Failed to read seapp hash for /data/user_de/0
```

```
E cutils : Failed to mkdir(/data/user_de/0/com.android.cts.priv.ctsshim): No such file or directory
E installd: Failed to prepare /data/user_de/0/com.android.cts.priv.ctsshim: No such file or directory
```

After that zygote is crashes. xD

**Comment by hungximbex** [comment]
@tank0412 , I build android 7.0 CM14 for zenfone 5 error kernel

**Comment by tank0412** [comment]
Bootanimation is passed.
Stuck on "Android is upgrading"

**Comment by dgadelha** [comment]
Anythong strange on logcat?

**Comment by tank0412** [comment]
W AudioSystem: AudioPolicyService not published, waiting...
I ServiceManager: Waiting for service media.audio_policy...

**Comment by dgadelha** [comment]
Do you have `/system/etc/audio_policy.conf`?

**Comment by tank0412** [comment]
@dgadelha It is not so easy.
Sure, i have it.

**Comment by dgadelha** [comment]
I know it is not. I haven't had time to work on it but I could analyze the logs later.

**Comment by tank0412** [comment]
Full logcat on Pastebin:
http://pastebin.com/WrnNLhpm

**Comment by tank0412** [comment]
@dgadelha  I know why it did not boot.
It did not boot due to libcamera_client.so.
NOTE: **This lib was built from AOSP N source**.
I swear that this lib IS NOT PROPRIEARY.
```
CANNOT LINK EXECUTABLE "mediaserver": cannot locate symbol "_ZNK7android8String1
64sizeEv" referenced by "/system/lib/libcamera_client.so".
```
Mediaserver did not start due to it.
We must shim missing symbols.

**Comment by dgadelha** [comment]
Simply delete it. :)
Also find the Android.mk and delete it

**Comment by tank0412** [comment]
I think that it is impossible because mediasserver call this lib anyway.
So if i delete it, it will crash because this lib is missing.
It is a dependency.
I guess this dependency can be removed from mediaserver source.
I have another solution. Android source must be patched for every missing symbol.
It works but it is calling a million of missing symbols.

**Comment by dgadelha** [comment]
I see your commits. You're not adding the missing symbols as I did in my previous `camera.patch`, instead you're creating empty voids. That's an obvious crash.

Still, you can boot android without a mediaserver. As I said, the important is booting, not booting with every single feature working.

**Comment by tank0412** [comment]
You saw my old and outdated commits.
Now i did this:
I patched frameworks/native/libs/binder/IPCThreadState.cpp.
I added this:
extern int _ZNK7android8String164sizeEv(IPCThreadState *state) {
    return state->getCallingUid();
}

extern int _ZNK7android7String86lengthEv(IPCThreadState *state) {
    return state->getCallingUid();
}

Need feedback of testers to check does it work.

**Comment by dgadelha** [comment]
13h ago is outdated? https://github.com/zf5/android_device_asus_ctp-common/commit/2bbe96630beb4908b96fe331e0cbcc2bd10893e8

And well, I'm without time to work on this.
Once I get some I'll take a look.
